nautilus_core/common/src/timer.rs

```diff
@@ -342,9 +342,11 @@ impl LiveTimer {
     /// Cancels the timer (the timer will not generate an event).
     pub fn cancel(&mut self) -> anyhow::Result<()> {
         debug!("Cancel timer '{}'", self.name);
-        if let Some(sender) = self.canceler.take() {
-            // Send cancellation signal
-            sender.send(()).map_err(|e| anyhow::anyhow!("{:?}", e))?;
+        if !self.is_expired.load(atomic::Ordering::SeqCst) {
```

Comment:
Does this actually fix the errors? because `self.canceler.take()` will take the canceler out of the option leaving behind `None` - so when its called again `sender` won't be Some.

Reply:
I just realized that the select could exit with `is_expired` set `True` before the timer is canceled, so also checking this `is_expired` value before going for the canceler would cover this.

Reply:
Hmm strange. For me the error doesn’t happen again, but have not seen the actual error message yet. I’ll dive deeper

Reply:
Ah indeed!

Reply:
I’ll make the change and test it. Somewhat later today.

Reply:
I think your impl is good as it is - especially if you don't see the errors anymore.

Reply:
Okay sure. Feel free to merge then if this is fine.